nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -4,6 +4,15 @@ with lib;
 
 let
 
+  sshconf = pkgs.runCommand "sshd.conf-validated" { } ''
+    cat >$out <<EOL
+    ${cfg.extraConfig}
+    EOL
+
+    ${cfgc.package}/bin/ssh-keygen -f mock-hostkey -N ""
+    ${cfgc.package}/bin/sshd -t -f $out -h mock-hostkey
```

Comment:
There are probably other modules doing the same but it just occurred to me that this will sadly not work if you cross-compile a NixOS system as these commands are to be executed at build time. Not sure if we are even able to get the correct binaries for the build platform given that the openssh package to use at runtime is configurable as a NixOS option.

Reply:
That's indeed a good point! How about adding a another option `services.openssh.validationPackage` that defaults to `cfg.package`, but can be modified when cross-building? Another option would be to simply allow turning the validation off I guess.